File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution(object):
    def  pallindrome(self,left,right,s):
        length=0
        while left>=0 and right<len(s) and s[left]==s[right]:
            length=right-left+1
            left=left-1
            right=right+1
        return length    
    def longestPalindrome(self, s):
        maxi=0
        l=0
        for i in range(len(s)):
            l1=self.pallindrome(i,i,s)
            l2=self.pallindrome(i,i+1,s)
            l=max(l1,l2)
            if l>maxi:
                maxi=max(maxi,l)
                start=i-(l-1)/2
        return    s[start:start+maxi]



        """
        :type s: str
        :rtype: str
        """
        

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
int pallindrome(int i,int j,string &s){
    int len=0;
    while(i>=0 && j<s.size() && s[i]==s[j]){
        len=j-i+1;
        i--;
        j++;
    }
    return len;
}
string longestPalindrome(string s) { 
    int a=0,max1=0;
    for(int i=0;i<s.size();i++){
        int l1=pallindrome(i,i,s);
        int l2=pallindrome(i,i+1,s);
        int l=max(l1,l2);
        if(l>max1){
            max1=l;
            a=i;
        }

    }
    return s.substr(a-(max1-1)/2,max1);
  


    }
};